apps/music/test/marionette/metadata_test.js

```diff
@@ -0,0 +1,73 @@
+/* global require, marionette, setup, suite, test, __dirname */
+'use strict';
+
+var assert = require('assert');
+var Music = require('./lib/music.js');
+var FakeRingtones = require('./lib/fakeringtones.js');
+var FakeControls = require('./lib/fakecontrols.js');
+var ListviewHelper = require('./lib/listviewhelper.js');
+
+marionette('Music player playlist', function() {
+  var apps = {};
+  apps[FakeRingtones.DEFAULT_ORIGIN] = __dirname + '/fakeringtones';
+  apps[FakeControls.DEFAULT_ORIGIN] = __dirname + '/fakecontrols';
+
+  var client = marionette.client({
+    prefs: {
+      'device.storage.enabled': true,
+      'device.storage.testing': true,
+      'device.storage.prompt.testing': true
+    },
+
+    settings: {
+      'lockscreen.enabled': false,
+      'ftu.manifestURL': null
+    },
+
+    apps: apps
+  });
+
+  var music;
+
+  setup(function() {
+    music = new Music(client);
+
+  });
+
+  suite('Album art display', function () {
+    setup(function() {
+      client.fileManager.removeAllFiles();
+      client.fileManager.add([
+        {
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/a.ogg'
+        }
+      ]);
+    });
+
+
+    test('Check album art is displayed', function() {
+
+      music.launch();
+      music.waitForFirstTile();
+
+      music.switchToArtistsView();
+
+      var bg = ListviewHelper.backgroundImageForListItem(music.listItems, 0);
+      // Test that the album cover is a blob. This mean it comes
+      // from the metadata.
+      assert.ok(bg.indexOf('url("blob:') != -1);
+
+      music.switchToAlbumsView();
+
+      bg = ListviewHelper.backgroundImageForListItem(music.listItems, 0);
+      assert.ok(bg.indexOf('url("blob:') != -1);
+
+      music.switchToSongsView();
+
+      bg = ListviewHelper.backgroundImageForListItem(music.listItems, 0);
+      assert.ok(bg.indexOf('url("blob:') != -1);
+    });
+
```

Comment:
Currently the art in the sample file is ````apps/music/test-data/album-art.jpg````. So we could perform an image comparison.
Indeed it make sense to check with a second file that has different art. I am not sure if the blob URL would be different for the same image or not. If there is no art we don't have a blob URL but rather a file URL.

I'll rework the test to improve this the way you suggested.